From 49fb992cd7bf94f25b93d9eeb711225daefa3a2d Mon Sep 17 00:00:00 2001
From: Oliver Smith <osmith@sysmocom.de>
Date: Mon, 21 Jan 2019 16:32:46 +0100
Subject: [PATCH] gsm_23_003.h: add GSM23003_IMEI_NUM_DIGITS_NO_CHK

Add new define for the 14 digit IMEI without the Luhn checksum, as it
is used in OsmoHLR.

Change-Id: I02b54cf01a674a1911c5c897fbec02240f88b521
---
 include/osmocom/gsm/protocol/gsm_23_003.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/osmocom/gsm/protocol/gsm_23_003.h b/include/osmocom/gsm/protocol/gsm_23_003.h
index ee697ff4..babd0f4b 100644
--- a/include/osmocom/gsm/protocol/gsm_23_003.h
+++ b/include/osmocom/gsm/protocol/gsm_23_003.h
@@ -24,6 +24,9 @@
 					 GSM23003_IMEI_SNR_NUM_DIGITS + 1)
 #define GSM23003_IMEISV_NUM_DIGITS	(GSM23003_IMEI_TAC_NUM_DIGITS + \
 					 GSM23003_IMEI_SNR_NUM_DIGITS + 2)
+/* IMEI without Luhn checksum */
+#define GSM23003_IMEI_NUM_DIGITS_NO_CHK	(GSM23003_IMEI_TAC_NUM_DIGITS + \
+					 GSM23003_IMEI_SNR_NUM_DIGITS)
 
 /* Chapter 19.2 "epc.mnc000.mcc000.3gppnetwork.org" */
 #define GSM23003_HOME_NETWORK_DOMAIN_LEN	33
-- 
2.16.4

